ui/src/conceptList.tsx

```diff
@@ -0,0 +1,103 @@
+import Typography from "@mui/material/Typography";
+import { DataGrid, GridColDef } from "@mui/x-data-grid";
+import React, { useContext, useEffect, useMemo, useState } from "react";
+import { useParams } from "react-router-dom";
+import { EntityInstancesApiContext } from "./apiContext";
+import { State } from "./state";
+
+// Row acts as a bridge between the data format returned from the API and the
+// format expected by DataGrid.
+class Row {
+  id = 0;
+  [x: string]: string | number | boolean | undefined;
+}
+
+type ConceptListProps = {
+  underlay: string;
+  state: State;
```

Comment:
Putting something called `State` into props seems slightly confusing to me since these are usually props and state are contrasting concepts in react.